File: backend/src/main/java/com/example/project/service/ChatService.java
package com.example.project.service;

import java.math.BigDecimal;
import java.util.List;

import com.example.project.dto.response.ChatMessageResponse;
import com.example.project.dto.response.ConversationResponse;

public interface ChatService {

    ConversationResponse getOrCreateConversation(String email, Long listingId);

    ConversationResponse getOrCreateRoommateConversation(String email, Long profileId);

    List<ConversationResponse> listMyConversations(String email);

    long getUnreadCount(String email);

    ConversationResponse getConversation(String email, Long conversationId);

    List<ChatMessageResponse> getMessages(String email, Long conversationId);

    ChatMessageResponse sendTextMessage(String email, Long conversationId, String content);

    ChatMessageResponse createDepositRequest(String email, Long conversationId, BigDecimal amount);
}

File: frontend/src/services/chatService.js
import api from './api';

export function getOrCreateConversation(listingId) {
  return api.post(`/chat/conversations/${listingId}`).then((res) => res.data);
}

export function getOrCreateRoommateConversation(profileId) {
  return api.post(`/chat/roommate-conversations/${profileId}`).then((res) => res.data);
}

export function getMyConversations() {
  return api.get('/chat/conversations').then((res) => res.data);
}

export function getConversation(conversationId) {
  return api.get(`/chat/conversations/${conversationId}`).then((res) => res.data);
}

export function getUnreadCount() {
  return api.get('/chat/unread-count').then((res) => res.data);
}

export function getMessages(conversationId) {
  return api.get(`/chat/conversations/${conversationId}/messages`).then((res) => res.data);
}

export function sendMessage(conversationId, content) {
  return api.post(`/chat/conversations/${conversationId}/messages`, { content }).then((res) => res.data);
}

export function createDepositRequest(conversationId, amount) {
  return api
    .post(`/chat/conversations/${conversationId}/deposit-request`, { amount })
    .then((res) => res.data);
}
